ports/esp32s2/i2s_common.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
```

Comment:
Why factor this out? The common-hal file looks pretty empty.

Reply:
Initially, before I knew that i2s could NOT drive the dacs in esp32s2, I hoped this code would support both digital and analog audio out.  I think there's some chance it could still support pdmin (using i2s-to-pdm algorithm similar to samd; esp32 supports pdm "directly" (hardware decimator) but again -s2 doesn't) or even i2sin.

Reply:
```components/soc/soc/esp32s2/include/soc/i2s_caps.h:#define SOC_I2S_SUPPORTS_PDM       (0) // ESP32-S2 don't support PDM
components/soc/soc/esp32s2/include/soc/i2s_caps.h:#define SOC_I2S_SUPPORTS_ADC_DAC   (0) // ESP32-S2 don't support ADC and DAC
```

Reply:
This might still be shared by I2SIn or PDMIn (if implemented as I2SIn + software postprocessing/filtering, like atmel-sam), but not for DAC-based AudioOut, so I moved it into common-hal/audiobusio.  Thanks!